core/src/main/java/org/elasticsearch/index/query/OrQueryBuilder.java

```diff
@@ -74,7 +98,66 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
     }
 
     @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        if (filters.isEmpty()) {
+            // no filters provided, this should be ignored upstream
+            return null;
+        }
+
+        BooleanQuery query = new BooleanQuery();
+        for (QueryBuilder f : filters) {
+            query.add(f.toQuery(parseContext), Occur.SHOULD);
```

Comment:
Can confirm this, if in the parser innerParser returns null, it's not added to the list of QueryBuilders we use. That logic is already in the parser code.